docs/csharp/roslyn-sdk/source-generators-overview.md

```diff
@@ -102,51 +84,48 @@ In this guide, you'll explore the creation of a source generator using the <xref
     }
     ```
 
-6. Replace the contents of the `Execute` method, so that it looks like the following:
+    A source generator needs to both implement the <xref:Microsoft.CodeAnalysis.ISourceGenerator?displayProperty=nameWithType> interface, and have the <xref:Microsoft.CodeAnalysis.GeneratorAttribute?displayProperty=nameWithType>. Not all source generators require initialization, and that is the case with this example implemenation &mdash; where <xref:Microsoft.CodeAnalysis.ISourceGenerator.Initialize%2A?displayProperty=nameWithType> is empty.
 
-    ```csharp
-    public void Execute(GeneratorExecutionContext context)
-    {
-        // find the main method
-        var mainMethod = context.Compilation.GetEntryPoint(context.CancellationToken);
-
-        // build up the source code
-        string source = $@"
-    using System;
-
-    namespace {mainMethod.ContainingNamespace.ToDisplayString()}
-    {{
-        public static partial class {mainMethod.ContainingType.Name}
-        {{
-            static partial void HelloFrom(string name)
-            {{
-                Console.WriteLine($""Generator says: Hi from '{{name}}'"");
-            }}
-        }}
-    }}
-    ";
-        // add the source code to the compilation
-        context.AddSource("generatedSource", source);
-    }
-    ```
+1. Replace the contents of the <xref:Microsoft.CodeAnalysis.ISourceGenerator.Execute%2A?displayProperty=nameWithType> method, with the following implementation:
+
+    :::code source="snippets/source-generators/SourceGenerator/HelloSourceGenerator.cs":::
+
+    From the `context` object we can access the compilations' entry point, or `Main` method. The `mainMethod` instance is an <xref:Microsoft.CodeAnalysis.IMethodSymbol>, and it represents a method or method-like symbol (including constructor, destructor, operator, or property/event accessor). From this object we can reason about the containing namespace (if one is present) and the type. The `source` in this example is an interpolated string that templates the source code to be generated, where the interpolated wholes are filled with the containing namespace and type information. The `source` is added to the `context` with a hint name.
+
+    > [!TIP]
+    > The `hintName` parameter from the <xref:Microsoft.CodeAnalysis.GeneratorExecutionContext.AddSource%2A?displayProperty=nameWithType> method can be any unique name, however; it is common to provide an explicit C# file extension such as `".g.cs"` or `".generated.cs"`. This will cause the generated file name to further help identify itself as being source generated.
```

Comment:
nit: I think readability (acro suggestions) this can be improved a little
```suggestion
    > The `hintName` parameter from the <xref:Microsoft.CodeAnalysis.GeneratorExecutionContext.AddSource%2A?displayProperty=nameWithType> method can be any unique name. It's common to provide an explicit C# file extension such as `".g.cs"` or `".generated.cs"` for the name. The file name helps identify the file as being source generated.
```